src/test/java/org/springframework/cloud/scheduler/spi/kubernetes/KubernetesSchedulerPropertiesTests.java

```diff
@@ -103,6 +104,61 @@ public void testNamespaceCanBeCustomized() {
 					StringUtils.hasText(kubernetesSchedulerProperties.getNamespace()));
 			assertEquals("Unexpected namespace", "myns", kubernetesSchedulerProperties.getNamespace());
 		}
+
+		@Test
+		public void testImagePullSecretDefault() {
+			KubernetesSchedulerProperties kubernetesSchedulerProperties = new KubernetesSchedulerProperties();
+			assertNull("No default image pull secret should be set",
+					kubernetesSchedulerProperties.getImagePullSecret());
+		}
+
+		@Test
+		public void testImagePullSecretCanBeCustomized() {
+			String secret = "mysecret";
+			KubernetesSchedulerProperties kubernetesSchedulerProperties = new KubernetesSchedulerProperties();
+			kubernetesSchedulerProperties.setImagePullSecret(secret);
+			assertNotNull("Image pull secret should not be null", kubernetesSchedulerProperties.getImagePullSecret());
+			assertEquals("Unexpected image pull secret", secret, kubernetesSchedulerProperties.getImagePullSecret());
+		}
+
+		@Test
+		public void testEnvironmentVariablesDefault() {
+			KubernetesSchedulerProperties kubernetesSchedulerProperties = new KubernetesSchedulerProperties();
+			assertEquals("No default environment variables should be set", 0,
+					kubernetesSchedulerProperties.getEnvironmentVariables().length);
+		}
+
+		@Test
+		public void testEnvironmentVariablesCanBeCustomized() {
+			String[] envVars = new String[] { "var1=val1", "var2=val2" };
+			KubernetesSchedulerProperties kubernetesSchedulerProperties = new KubernetesSchedulerProperties();
+			kubernetesSchedulerProperties.setEnvironmentVariables(envVars);
+			assertNotNull("Environment variables should not be null",
+					kubernetesSchedulerProperties.getEnvironmentVariables());
+			assertEquals("Unexpected number of environment variables", 2,
+					kubernetesSchedulerProperties.getEnvironmentVariables().length);
+		}
+
+		@Test
+		public void testTaskServiceAccountNameDefault() {
+			KubernetesSchedulerProperties kubernetesSchedulerProperties = new KubernetesSchedulerProperties();
+			assertNotNull("Task service account name should not be null",
+					kubernetesSchedulerProperties.getTaskServiceAccountName());
+			assertEquals("Unexpected default task service account name",
+					KubernetesSchedulerProperties.DEFAULT_TASK_SERVICE_ACCOUNT_NAME,
+					kubernetesSchedulerProperties.getTaskServiceAccountName());
+		}
+
+		@Test
+		public void testTaskServiceAccountNameCanBeCustomized() {
+			String taskServiceAccountName = "mysa";
+			KubernetesSchedulerProperties kubernetesSchedulerProperties = new KubernetesSchedulerProperties();
+			kubernetesSchedulerProperties.setTaskServiceAccountName(taskServiceAccountName);
+			assertNotNull("Task service account name should not be null",
```

Comment:
You don't have to use it now.  But AssertJ makes these types of checks alot smoother. 
For Example:
https://github.com/spring-cloud/spring-cloud-task/blob/master/spring-cloud-task-core/src/test/java/org/springframework/cloud/task/SimpleTaskConfigurationTests.java#L77
https://github.com/spring-cloud/spring-cloud-task/blob/master/spring-cloud-task-core/src/test/java/org/springframework/cloud/task/SimpleTaskConfigurationTests.java#L81

Reply:
nice thanks - old habits die hard ;-) theres also some nice assertions out of the fabric8 libs, but again..